packages/hint-manifest-icons/src/hint.ts

```diff
@@ -267,6 +270,73 @@ export default class ManifestIconHint implements IHint {
             return validSizes;
         };
 
+        const validatePurposesValues = (purposes: string[], resource: string, iconIndex: number, getLocation: JSONLocationFunction) => {
+            for (const purpose of purposes) {
+                // Check if the purpose is valid.
+                if (!validPurposes.includes(purpose)) {
+                    context.report(resource, getMessage('iconPurposeInvalid', context.language, purpose), {
+                        location: getLocation(`icons[${iconIndex}].purpose`),
+                        severity: Severity.warning
+                    });
+                }
+            }
+        };
+
+        const validateDuplicatedValues = (purposes: string[], resource: string, iconIndex: number, getLocation: JSONLocationFunction) => {
+            /*
+             * Creating a set will remove automatically the duplicated items.
+             */
+            const purposesSet = new Set(purposes);
+
+            if (purposes.length === purposesSet.size) {
+                // No duplicated found.
+                return;
+            }
+
+            const purposesCopy = [...purposes];
+
+            for (const purpose of purposesSet) {
+                const purposeIndex = purposesCopy.indexOf(purpose);
+
+                if (purposeIndex >= 0) {
+                    purposesCopy.splice(purposeIndex, 1);
+                }
+            }
+
+            /*
+             * Check if purposesCopy contains any purpose, if so, those
+             * items are duplicated.
+             */
+            if (purposesCopy.length > 0) {
+                context.report(resource, getMessage('iconPurposeDuplicate', context.language, [...(new Set(purposesCopy))].join(' ')), {
+                    location: getLocation(`icons[${iconIndex}].purpose`),
+                    severity: Severity.hint
+                });
+            }
+        };
+
+        // See https://w3c.github.io/manifest/#purpose-member
+        const validateIconsPurpose = (icons: ManifestImageResource[], resource: string, getLocation: JSONLocationFunction) => {
+            for (let iconIndex = 0; iconIndex < icons.length; iconIndex++) {
+                const icon = icons[iconIndex];
+
+                if (!icon.purpose) {
+                    continue;
+                }
+
+                /*
+                 * Remove unnecessary empty spaces before split the array.
+                 */
+                const purposes = icon.purpose
+                    .trim()
+                    .replace(/\s\s*/g, ' ')
+                    .split(' ');
```

Comment:
We can just pass the regex to `split` directly.
```suggestion
                    .split(/\s\s*/g);
```